File: atcoder.jp/Python (3.8.2)/abc183/abc183_d.py
N, W = map(int, input().split())

water = [0] * (2 * 10 ** 5 + 2)

for _ in range(N):
    S, T, P = map(int, input().split())
    water[S] += P
    water[T] -= P

for i in range(2 * 10 ** 5 + 1):
    water[i+1] += water[i]

if all(w <= W for w in water):
    print("Yes")
else:
    print("No")

File: atcoder.jp/Rust (rustc 1.70.0)/abc183/abc183_d.rs
use proconio::{input, fastout};

#[fastout]
fn main() {
    input! {
        n: usize,
        w: isize,
    }

    let mut vec = vec![0; 2 * 100_000 + 1];

    for _ in 0..n {
        input! {
            s: usize,
            t: usize,
            p: isize,
        }

        vec[s] += p;
        vec[t] -= p;
    }

    for i in 1..vec.len() {
        vec[i] += vec[i - 1];
    }

    let ans = vec.into_iter().all(|x| x <= w);
    if ans {
        println!("Yes");
    } else {
        println!("No");
    }
}
